app/src/main/java/io/apicurio/registry/storage/impl/sql/MySQLSqlStatements.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.storage.impl.sql;
+
+/**
+ * MySQL implementation of the sql statements interface.  Provides sql statements that
+ * are specific to MySQL, where applicable.
+ * @author bruno.ariev@gmail.com
+ */
+public class MySQLSqlStatements extends CommonSqlStatements {
+
+    /**
+     * Constructor.
+     */
+    public MySQLSqlStatements() { }
+
+    /**
+     * @see SqlStatements#dbType()
+     */
+    @Override
+    public String dbType() {
+        return "mysql";
+    }
+
+    /**
+     * @see SqlStatements#isPrimaryKeyViolation(Exception)
+     */
+    @Override
+    public boolean isPrimaryKeyViolation(Exception error) {
+        return error.getMessage().contains("violates unique constraint");
+    }
+
+    /**
+     * @see SqlStatements#isForeignKeyViolation(Exception)
+     */
+    @Override
+    public boolean isForeignKeyViolation(Exception error) {
+        return error.getMessage().contains("violates foreign key constraint");
+    }
+
+    /**
+     * @see SqlStatements.core.storage.jdbc.ISqlStatements#isDatabaseInitialized()
+     */
+    @Override
+    public String isDatabaseInitialized() {
+        return String.join(" ",
+                "SELECT COUNT(*) AS count",
+                "FROM information_schema.tables",
+                "WHERE table_name = 'artifacts';");
+    }
+
+    /**
+     * @see SqlStatements#upsertContent()
+     */
+    @Override
+    public String upsertContent() {
+        return String.join(" ",
+                "INSERT IGNORE INTO content",
+                "(tenantId, contentId, canonicalHash, contentHash, content, artifactreferences)",
+                "VALUES (?, ?, ?, ?, ?, ?);");
+    }
+
+    /**
+     * @see SqlStatements#upsertLogConfiguration()
+     */
+    @Override
+    public String upsertLogConfiguration() {
+        return String.join(" ",
+                "INSERT INTO logconfiguration (logger, loglevel)",
+                "VALUES (?, ?)",
+                "ON DUPLICATE KEY UPDATE loglevel = ?;");
+    }
+
+    /**
+     * @see SqlStatements#getNextSequenceValue()
+     */
+    @Override
+    public String getNextSequenceValue() {
+        //TODO: Create a Stored Procedure to return the VALUE after the UPSERT
```

Comment:
Are you sure you need a stored procedure here?  Seems overkill.

Reply:
I've got to test this some more, I took the MySQL and SQLServer implementation as the basis and MySQL doesn't have the equivalent: 
Postgres
```
    @Override
    public String getNextSequenceValue() {
        return "INSERT INTO sequences (tenantId, name, value) VALUES (?, ?, 1) ON CONFLICT (tenantId, name) DO UPDATE SET value = sequences.value + 1 RETURNING value";
    }
```  

SQLServer:
```
    @Override
    public String getNextSequenceValue() {
        return String.join(" ",
                "MERGE INTO sequences AS target",
                "USING (VALUES  (?, ?)) AS source (tenantId, name)",
                "ON (target.tenantId = source.tenantId AND target.name = source.name)",
                "WHEN MATCHED THEN",
                    "UPDATE SET value = target.value + 1",
                "WHEN NOT MATCHED THEN",
                    "INSERT (tenantId, name, value)",
                    "VALUES (source.tenantId, source.name, 1)",
                "OUTPUT INSERTED.value;");
    }
```